WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -564,12 +642,23 @@ public void onMediaUploaded(OnMediaUploaded event) {
 
                                 // Replace local with remote media in the post content
                                 PostModel updatedPost = updatePostWithCurrentlyCompletedUploads(latestPost);
+                                // also do the same now with failed uploads
+                                updatedPost = updatePostWithCurrentlyFailedUploads(updatedPost);
+                                // finally, save the PostModel
                                 mDispatcher.dispatch(PostActionBuilder.newUpdatePostAction(updatedPost));
 
                                 // TODO Should do some extra validation here
                                 // e.g. what if the post has local media URLs but no pending media uploads?
-                                iterator.remove();
-                                mPostUploadHandler.upload(updatedPost);
+
+                                // only upload Post if there are no failed media as well
+                                if (!hasAnyFailedMediaUploadsForPost(uploadingPost.postModel)) {
```

Comment:
It doesn't look like it's possible for `hasAnyFailedMediaUploadsForPost()` to be `true` at this point - if this post had failed media, we'd have already removed it from `sPostsWithPendingMedia` when we received a cancelled or errored `OnMediaUploaded` event. What do you think about simplifying this logic and using what was there before?